### PR TITLE
Added BlockProcessorCapella to support Capella changes in spec logic.

### PR DESCRIPTION
**Pull request by rolfyone**
- Created some of the helper classes

 - removed reference to TransitionHelpers which were Bellatrix specific

Signed-off-by: Paul Harris <paul.harris@consensys.net>

## Documentation

- [X] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [X] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

**Review by ajsutton — COMMENTED** [review]
I suspect we've gone too far here. We should just keep using the `Bellatrix` (or even earlier) versions of things unless there's actually something to change. Most of these new classes won't be needed.

I suspect now is the time we want to stop just adding infrastructure and starting letting the actual implementation drive what's required.

File: ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/versions/capella/forktransition/CapellaStateUpgrade.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+              state.setFork(
+                  new Fork(
+                      preState.getFork().getCurrentVersion(),
+                      specConfig.getCapellaForkVersion(),
+                      epoch));
+
+              state.setLatestExecutionPayloadHeader(
+                  schemaDefinitions.getExecutionPayloadHeaderSchema().getDefault());
+              state.setLatestWithdrawalValidatorIndex(UInt64.valueOf(state.getValidators().size()));
```
Comment (ajsutton) [inline ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/versions/capella/forktransition/CapellaStateUpgrade.java]: This is actually initialised to 0: https://github.com/ethereum/consensus-specs/pull/3068/files#diff-c06c6116e0fa7993a0d4d7925b64cd2e5f7a23015a93e3c4d8d51209fe9f4dc0R133

File: ethereum/spec/src/test/java/tech/pegasys/teku/spec/logic/versions/capella/BlockProcessorCapellaTest.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+import tech.pegasys.teku.spec.util.DataStructureUtil;
+
+public class BlockProcessorCapellaTest extends BlockProcessorBellatrixTest {
+  @Override
+  protected Spec createSpec() {
+    return TestSpecFactory.createMainnetCapella();
+  }
+
+  @Test
+  void shouldIgnoreBellatrixBlock() {
```
Comment (ajsutton) [inline ethereum/spec/src/test/java/tech/pegasys/teku/spec/logic/versions/capella/BlockProcessorCapellaTest.java]: nit: shouldRejectBellatrixBlock? And probably rename `shouldIgnoreAltairBlock` while we're at it.

**Review by ajsutton — APPROVED** [review]
LGTM, but need to get the initial value for `latest_withdrawal_validator_index` right.

File: ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/versions/capella/forktransition/CapellaStateUpgrade.java (hunk 1 above)
Comment (rolfyone) [reply ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/versions/capella/forktransition/CapellaStateUpgrade.java]: oh so it misses. the first index, thats sad, but ok

File: ethereum/spec/src/test/java/tech/pegasys/teku/spec/logic/versions/capella/BlockProcessorCapellaTest.java (hunk 2 above)
Comment (rolfyone) [reply ethereum/spec/src/test/java/tech/pegasys/teku/spec/logic/versions/capella/BlockProcessorCapellaTest.java]: fair

**Pull request merged**